### PR TITLE
bump re2 for wasm build

### PR DESCRIPTION
**Pull request by Shikugawa**
Signed-off-by: Shikugawa <rei@tetrate.io>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -433,9 +433,9 @@ DEPENDENCY_REPOSITORIES = dict(
         cpe = "N/A",
     ),
     com_googlesource_code_re2 = dict(
-        sha256 = "04ee2aaebaa5038554683329afc494e684c30f82f2a1e47eb62450e59338f84d",
-        strip_prefix = "re2-2020-03-03",
-        urls = ["https://github.com/google/re2/archive/2020-03-03.tar.gz"],
+        sha256 = "455bcacd2b94fca8897decd81172c5a93e5303ea0e5816b410877c51d6179ffb",
+        strip_prefix = "re2-2b25567a8ee3b6e97c3cd05d616f296756c52759",
+        urls = ["https://github.com/google/re2/archive/2b25567a8ee3b6e97c3cd05d616f296756c52759.tar.gz"],
```
Comment (asraa) [inline bazel/repository_locations.bzl:438]: Can you please add the date for this commit above this line?

Comment (Shikugawa) [reply bazel/repository_locations.bzl:439]: done

Comment (moderation) [reply bazel/repository_locations.bzl:439]: And a TODO to switch back to a release version when your commit is inluded. @junyer generally releases monthly on the first of the month.

Comment (Shikugawa) [reply bazel/repository_locations.bzl:442]: added a TODO comment.

**Pull request merged**